### ISSUE TITLE
[Feature Request]: Force-install apps that are on allowlist

### ISSUE THREAD

**Issue by AlexJPawlak** [Romanitho/Winget-AutoUpdate#760]
### The request

I think it would be great addon if the app, when operating in "allowlist mode" had a switch to force installation if the app is not detected, and have that option configurable in the Allowlist setting - for instance when we allowlist 7zip updates, we give the package name - 7zip.7zip;1 to enable autoinstall, or 7zip.7zip;0 (or default) to disable autoinstall (current behaviour) 

### Additional information

My usecase is that I want to make sure select safe apps are always updated, and ideally I want to make sure they are also installed on the box I'm managing - 7zip being one instance of such app. There are number of others, but this solution would give me a nice cut in to install apps. Currently I'm wrapping some home baked winget script to do this thing, but your solution seem much more mature and better suited as overall "Winget Manager"

**Comment by Romanitho** [comment]
not sure to understand your point. if an app is in the white list, it is updated. Installations are nothing related to that.

**Comment by myrandor** [comment]
I have the same request. I have a list of app that I want to install on every computer. The allow list only restrict updatea on some apps, but won't install them if they are not installed.

**Comment by AlexJPawlak** [comment]
@Romanitho So, if you are using an allowlist, WAU will attempt to run upgrade command. However, the upgrade command will not do anything if the package is not installed at all on the machine. With the proposed change, a new feature flag would be introduced that will instrument WAU to run "winget install" if it's not detected, so that every device that gets that policy, would also install the requested package

**Comment by AndrewDemski-ad-gmail-com** [comment]
## TL;DR:
This would be a cool feature in the non-existent Winget-AutoInstall tool buuut.. 
..that would be a counterproductive from software management perspective :)

## software management
### aspect1 - unmanaged deployment of SW
Decisions about what to install should not be made lightly. Every additional installation is a potential security risk.
This risk needs to be recognized, evaluated, assessed, and approved.
Last thing YOU would want to say to your Manager, Boss, friends and/or family would be "it was in winget so it should work on its own, I do not know why your AV showed warnings and now your computer does not work after reboot".

### aspect2 - vulnerabilities remediation
WAU offers a slightly simplified logic that primarily allows for patching software vulnerabilities and security holes, making life easier for those who previously patched things manually or using platform management solutions.
This is where it shines and how it is used by the vast majority of users.
WAU aims at reducing the vulnerabilities (by focusing on upgrading of existing SW versions), not introducing new ones.

### aspect3 - Application lifecycle control
It was generally accepted that products placed in the winget repository could be installed/updated using some unknown test scenarios, it does not meant that it would work everywhere else 100% of the time.
That is not enough to justify not checking "what is in the box". 
SBOMs/Software Bills Of Material are a thing for a reason.
WAU will never accept a licence which will turn on a paid subscription and force you to pay for anything.
That decision had to be mage in advance, and definitely not by WAU contributors.

### aspect4 - SW needs settings, not WYSIWYG
WAU does not offer the simple, often required customization before, during and after initial installation.
WAU would not care whether a given product is installed correctly, in the right order or not ASAP.
If something failed, WAU would retry the failing action the next day.. and the next.. etc. 

### aspect5 - changing requirements
In the case where WAU would install and then upgrade the product versions, the burden of properly customizing the installation would still be YOURS, because the risk of delivering software that is not tailored to the needs will not lie with the WAU creators.

Agreeing to the lazy in its nature suggestion that "someone else should install programs for me, because I don't feel like it, and I want some tool" would ultimately lead to the need to prepare customization in the form of MOD files for each installation.

**Comment by AlexJPawlak** [comment]
While I understand your points and they are fair, what happens if a winget package changes licensing? Say, 7zip goes from MIT to some proprietary license and prompts an update - will WAU fail or "accept" on our behalf? Also, I do think that if that feature is scoped to "Allowlist" configuration (which is not the default one), it would imply admins using this tool are aware of what they are doing and thus - understand or should understand the risks. Apparently - all your arguments can hold well against what WAU is currently doing. Technically speaking, an "upgrade" is just an installation of the software on top of existing software. All the proposed change would be about is to flag "Install the software regardless if its installed or not". Make this option configurable only by GPO or Registry and it will work well in most cases.

**Comment by AndrewDemski-ad-gmail-com** [comment]
@AlexJPawlak ,there is a relatively fresh PR which would allow WAU to experiment a bit.

https://github.com/Romanitho/Winget-AutoUpdate/pull/789

The main purpose of adding that experimental set of settings was providing a way of using a custom winget repository.

With such thing running in your ecosystem, the problem which could be potentially costly if WAU was able to install stuff using some pre-install list can be visually eliminated.

Would you be willing to help us in expanding that logic to make it more mature and safe?

**Comment by Romanitho** [comment]
For me, WAU was initially designed to automatically update apps. Over time, new features such as the White/Blacklist, Mods, etc., were implemented. However, using the Whitelist as a way to install apps (like deploying a master or something) was never part of the original scope and feels a bit risky to me.

In the meantime, the Winget-Install project, which is now integrated into WAU, can be used to install a few apps at once. Maybe you can use it to install the apps you need on your machines. 😊
